src/main/java/input_feed/distributed/JSONToDistributedR1CS.java

```diff
@@ -0,0 +1,196 @@
+            Object obj = parser.parse(new FileReader(filePath));
+
+            jsonObject = (JSONObject) obj;
+            constraintList = (JSONArray) jsonObject.get("constraints");
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assert (numConstraints == constraintList.size());
+
+        JSONArray[] constraintArray = new JSONArray[numConstraints];
+        for (int i = 0; i < numConstraints; i++) {
+            constraintArray[i] = (JSONArray) constraintList.get(i);
+        }
+
+        final ArrayList<Integer> partitions =
+                constructPartitionArray(config.numPartitions(), numConstraints);
+
+        // Load Linear Combinations as RDD format
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationA = distributedCombinationFromJSON(
+                partitions, constraintArray, 0, numConstraints, config);
+
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationB = distributedCombinationFromJSON(
+                partitions, constraintArray, 1, numConstraints, config);
+
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationC = distributedCombinationFromJSON(
+                partitions, constraintArray, 2, numConstraints, config);
+
+        combinationA.count();
+        combinationB.count();
+        combinationC.count();
+
+        final R1CSConstraintsRDD<FieldT> constraints = new R1CSConstraintsRDD<>(
+                combinationA, combinationB, combinationC, numConstraints);
+
+        return new R1CSRelationRDD<>(constraints, numInputs, numAuxiliary);
+    }
+
+    public Tuple2<Assignment<FieldT>, JavaPairRDD<Long, FieldT>> loadWitness(Configuration config) {
+        JSONParser parser = new JSONParser();
+        JSONObject jsonObject;
+        JSONArray primaryInputs = new JSONArray();
+        JSONArray auxInputs = new JSONArray();
+
+        try {
+            Object obj = parser.parse(new FileReader(filePath));
+
+            jsonObject = (JSONObject) obj;
+            primaryInputs = (JSONArray) jsonObject.get("primary_input");
+            auxInputs = (JSONArray) jsonObject.get("aux_input");
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        // TODO - Serial assignment may not be necessary.
+        final Assignment<FieldT> serialAssignment = new Assignment<>();
+        for (Object input: primaryInputs) {
+            final FieldT value = fieldParameters.construct((String) input);
+            serialAssignment.add(value);
+        }
+        for (Object input: auxInputs) {
+            final FieldT value = fieldParameters.construct((String) input);
+            serialAssignment.add(value);
+        }
+
+        final long numVariables = primaryInputs.size() + auxInputs.size();
+        final int numExecutors = config.numExecutors();
+        final ArrayList<Integer> assignmentPartitions = constructPartitionArray(numExecutors, numVariables);
+
+        JavaPairRDD<Long, FieldT> oneFullAssignment = config.sparkContext()
```

Comment:
Could this be refactored/simplified in a similar fashion as what we did for the TextToR1CS?